tests/components/homepluscontrol/test_integration.py

```diff
@@ -0,0 +1,794 @@
+"""Test the Legrand Home+ Control integration."""
+import asyncio
+from unittest.mock import patch
+
+from homepluscontrol.homeplusplant import (
+    PLANT_TOPOLOGY_BASE_URL,
+    PLANT_TOPOLOGY_RESOURCE,
+)
+
+from homeassistant import config_entries, setup
+from homeassistant.components.homepluscontrol import api
+from homeassistant.components.homepluscontrol.const import DOMAIN, PLANT_URL
+
+
+async def entity_assertions(
+    hass,
+    num_exp_entities,
+    num_exp_devices=None,
+    coordinator=None,
+    expected_entities=None,
+    expected_devices=None,
+):
+    """Assert number of entities and devices."""
+    entity_reg = await hass.helpers.entity_registry.async_get_registry()
+    device_reg = await hass.helpers.device_registry.async_get_registry()
+
+    if coordinator is None:
+        coordinator = hass.data["homepluscontrol"][
```

Comment:
We should not access integration details in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
Understood. I removed this part of the test code.